builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -33,14 +35,23 @@ func resourceAwsS3Bucket() *schema.Resource {
 
 func resourceAwsS3BucketCreate(d *schema.ResourceData, meta interface{}) error {
 	s3conn := meta.(*AWSClient).s3conn
+	awsRegion := meta.(*AWSClient).region
```

Comment:
It should pick up the region from the `provider` block in your configuration, yes:

- https://github.com/hashicorp/terraform/blob/116e6cac33e324a382f2cb5933d1faf6214a4bf5/builtin/providers/aws/config.go#L119